README.md

```diff
@@ -101,25 +101,41 @@ as part of a pipeline or inside a pheonix controller.
 defmodule MyApp.MyController do
   use MyApp.Web, :controller
 
-  plug Guardian.Plug.EnsureAuthenticated, on_failure: { MyApp.MyController, :unauthenticated }
+  plug Guardian.Plug.EnsureAuthenticated, handler: MyApp.MyAuthErrorHandler
 end
 ```
 
 The failure function must receive the connection, and the connection params.
 
+The error handler may be defined inline or in the main configuration.
+
+```elixir
+config :guardian, Guardian,
+  handler: MyApp.MyAuthErrorHandler
+  # …
+```
+
 ### Guardian.Plug.EnsurePermissions
 
 Looks for a previously verified token. If one is found, confirms that all listed
-permissions are present in the token. If not, the failure function is called.
+permissions are present in the token. If not, the `:unauthorized` function is called on your handler.
 
 ```elixir
 defmodule MyApp.MyController do
   use MyApp.Web, :controller
 
-  plug Guardian.Plug.EnsurePermissions, on_failure: { MyApp.MyController, :forbidden }, default: [:read, :write]
+  plug Guardian.Plug.EnsurePermissions, handler: MyApp.MyAuthErrorHandler, default: [:read, :write]
 end
 ```
 
+The error handler may be defined inline or in the main configuration.
+
+```elixir
+config :guardian, Guardian,
+  handler: MyApp.MyAuthErrorHandler
+  # …
```

Comment:
This is global, so I am :-1:  on it. The user can define their own handler module, then plug that anywhere they want.

```elixir
defmodule MyApp.EnsureAuthenticated do
  use Plug.Builder
  plug Guardian.Plug.EnsureAuthenticated, handler: __MODULE__

  def unauthenticated(conn) do
    ...
  end
end


defmodule MyApp.MyController do
  plug MyApp.EnsureAuthenticated
end
```